src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -477,15 +477,17 @@ class World internal constructor(
             return
         }
 
+        val versionLookup = snapshot.keys.associateBy { it.id }
```

Comment:
That's the part of my question from the other PR, correct? I still think that storing and restoring the entity version is not needed for snapshots. But even if it is, we'd need to reset the entity version at some point because otherwise it simply is increasing infinitly and will overflow at some point.

Reply:
I created a test to show why I think this is needed: https://github.com/Quillraven/Fleks/pull/115/files#diff-1deae4e923b3764d2c4e62b4862e043ab20d510fbdc80b777dc23fc25117cbfdR825
Currently with version as a Long, it would take 299'865'143 years of continuously recycling the same entity at 1000 frames per second to reach Long.MAX_VALUE. Twice that to wrap around back to version=0. At that point it would start to reuse versions which would only be a problem if the game actually still has a reference to that version and tries to use it.
If we changed version to Int, it would be about 24 days to reach Int.MAX_VALUE or 49 days to reuse version=0.

Given these estimates I think we could go down to storing the version as an Int and still ignore the issue that it is theoretically possible that a version gets reused.

Reply:
I prefer the `Int` version approach because as you mentioned `Long` is a little bit ridiculous and wastes more memory. Also it is consistent with `id` which is currently also an `Int`.

However, I didn`t get how we solve the overflow issue right now. Doesn`t the version become a negative number at some point and then break everything? I guess maybe not, because the version isn't used for any array index, correct? But still, wouldn't it be nicer to not allow negative versions and simply start over with 0 again?

Reply:
I remembered that Kotlin has unsigned integers, version is now an UInt ;-)